pytorch_lightning/trainer/trainer.py

```diff
@@ -1772,14 +1772,15 @@ def _log_device_info(self) -> None:
 
         if torch.cuda.is_available() and self._device_type != _AcceleratorType.GPU:
             rank_zero_warn(
-                "GPU available but not used. Set the gpus flag in your trainer `Trainer(gpus=1)` or script `--gpus=1`.",
+                "GPU available but not used. Set `accelerator` and `devices` via your"
+                " trainer `Trainer(accelerator='gpu', devices=1)`.",
                 category=PossibleUserWarning,
             )
 
         if _TPU_AVAILABLE and self._device_type != _AcceleratorType.TPU:
             rank_zero_warn(
-                "TPU available but not used. Set the `tpu_cores` flag in your trainer"
-                " `Trainer(tpu_cores=8)` or script `--tpu_cores=8`."
+                "TPU available but not used. Set `accelerator` and `devices` via your trainer"
+                " `Trainer(accelerator='tpu', devices=8)`."
```

Comment:
This warning should provide the number of available devices, not hardcode 8

Reply:
TPUs always come up with the set of 8 devices. But yes, we could update this for ipus, gpus, etc.

Also, when the user does `Trainer(accelerator="tpu")`, the devices are set to `auto_device_count`. That's the default.

Reply:
Updated!